src/full.rs

```diff
@@ -0,0 +1,157 @@
+use crate::{Body, SizeHint};
+use bytes::{Buf, Bytes};
+use http::HeaderMap;
+use std::borrow::Cow;
+use std::convert::{Infallible, TryFrom};
+use std::marker::Unpin;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// A body that consists of a single chunk.
+#[derive(Clone, Copy, Debug)]
+pub struct Full<D> {
+    data: Option<D>,
+}
+
+impl<D> Full<D>
+where
+    D: Buf + Unpin,
```

Comment:
Do we need `Unpin`? I *think* since this never propagates the `Pin` to be `Pin<&mut D>`, then we don't need it. But it'd be worth to double check that...

Reply:
Turned out that we didn't need `Unpin`. I've removed the bounds.